lib/dns/zone.js

```diff
@@ -0,0 +1,817 @@
+  });
+};
+
+/**
+ * Provide a path to a zone file to copy records into the zone.
+ *
+ * @param {string} localPath - The fully qualified path to the zone file.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API or file system error.
+ *
+ * @example
+ * var zoneFilename = '/Users/stephen/zonefile.zone';
+ *
+ * zone.export(zoneFilename, function(err) {
+ *   if (!err) {
+ *     // The zone file was created successfully.
+ *   }
+ * });
+ */
+Zone.prototype.export = function(localPath, callback) {
+  this.getRecords(function(err, records) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var stringRecords = records.map(util.exec('toString')).join('\n');
+
+    fs.writeFile(localPath, stringRecords, 'utf-8', function(err) {
+      callback(err || null);
+    });
+  });
+};
+
+/**
+ * Get the list of changes associated with this zone. A change is an atomic
+ * update to a collection of records.
+ *
+ * @param {object=} query - The query object.
+ * @param {number} query.maxResults - Maximum number of results to return.
+ * @param {string} query.pageToken - The page token.
+ * @param {string} query.sort - Set to 'asc' for ascending, and 'desc' for
+ *     descending or omit for no sorting.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:dns/change[]} callback.changes - An array of
+ *     {module:dns/change} objects.
+ * @param {?object} callback.nextQuery - A query object representing the next
+ *     page of results.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var callback = function(err, changes, nextQuery, apiResponse) {
+ *   // The `metadata` property is populated for you with the metadata at the
+ *   // time of fetching.
+ *   changes[0].metadata;
+ *
+ *   // However, in cases where you are concerned the metadata could have
+ *   // changed, use the `getMetadata` method.
+ *   changes[0].getMetadata(function(err, metadata) {});
+
+ *   if (nextQuery) {
+ *     // nextQuery will be non-null if there are more results.
+ *     zone.getChanges(nextQuery, callback);
+ *   }
+ * };
+ *
+ * zone.getChanges(callback);
+ *
+ * //-
+ * // Get the changes from your zone as a readable object stream.
+ * //-
+ * zone.getChanges()
+ *   .on('error', console.error)
+ *   .on('data', function(change) {
+ *     // change is a Change object.
+ *   })
+ *   .on('end', function() {
+ *     // All changes retrieved.
+ *   });
+ *
+ * //-
+ * // If you anticipate many results, you can end a stream early to prevent
+ * // unnecessary processing and API requests.
+ * //-
+ * zone.getChanges()
+ *   .on('data', function(change) {
+ *     this.end();
+ *   });
+ */
+Zone.prototype.getChanges = function(query, callback) {
+  var self = this;
+
+  if (util.is(query, 'function')) {
+    callback = query;
+    query = {};
+  }
+
+  if (query.sort) {
+    query.sortBy = 'changeSequence';
```

Comment:
Do we need this? It looks like this is the default (and only) `sortBy` value upstream,